code/game/objects/structures/tall_stairs.dm

```diff
@@ -0,0 +1,175 @@
+/obj/structure/tall_stairs
+	name = "stairs"
+	icon = 'icons/obj/structures_tall.dmi'
+	icon_state = "stairs_start"
+	anchored = TRUE
+	move_resist = INFINITY
+
+	/// Our paired stairs, used for tracking destruction
+	var/obj/structure/tall_stairs/paired
+	/// Should we destroy our paired stairs?
+	var/destroy_paired = TRUE
+
+/obj/structure/tall_stairs/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/climb_walkable)
+	locate_pair()
+
+/obj/structure/tall_stairs/proc/locate_pair()
+	return
+
+/obj/structure/tall_stairs/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if (!isnull(paired))
+		paired.paired = null
+		paired = null
+	locate_pair()
+
+/obj/structure/tall_stairs/Destroy(force)
+	. = ..()
+	if (!QDELETED(paired) && destroy_paired)
+		paired.destroy_paired = FALSE
+		QDEL_NULL(paired)
+
+/obj/structure/tall_stairs/start
+	icon_state = "stairs_start"
+
+/obj/structure/tall_stairs/start/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/elevation, pixel_shift = 8)
```

Comment:
did you look into ghommie's thoughts on this? (Some way to smooth out the animation to match a slope/glide_size?)